### PR TITLE
Make sure caption controls "dim out" when window loses NC focus

### PR DESCRIPTION
**Pull request by AnuthaDev**
Smol Code

## Summary of the Pull Request
 
Make the caption controls "dim out" when window loses focus

## References
https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-ncactivate
https://github.com/microsoft/terminal/pull/5881
https://github.com/microsoft/terminal/pull/3025
#7015 

## PR Checklist
* [x] Applies to #1625
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments

Added a handler for WM_NCACTIVATE to propagate the focus change to the Titlebar Control.

The unfocused color is set to ForegroundBaseLow

## Validation Steps Performed

None

(Potato™ PC, no compile)

### CODE REVIEW

**Comment by AnuthaDev** [comment]
Phone: 1, Code-Format Checker: 0

**Comment by AnuthaDev** [comment]
BTW, I would love some guidance on how to compile just the minimum projects necessary for testing UI changes (preferably using VS)

**Comment by AnuthaDev** [comment]
अरे कोई रिव्यू कर दो

File: src/cascadia/TerminalApp/TitlebarControl.idl [hunk 1]
```diff
@@ -7,14 +7,18 @@ namespace TerminalApp
     {
         WindowVisualStateNormal = 0,
         WindowVisualStateMaximized,
-        WindowVisualStateIconified
+        WindowVisualStateIconified,
+        WindowVisualStateFocused,
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TitlebarControl.idl:11]: I'm still pretty worried about including these in the list of window states. I think they *need* to be a separate enum. Here's why:

The window is always aware of its current visual state, right...

What happens if you *maximize* it, then *un-focus it*, and then *focus it again*?

The visual state, according to the window, is no longer "maximized." It's "focused." It totally forgot that it was maximized!

We need to track the focus state separately from whether the window is normal, maximized or iconified.

**Review by DHowett — CHANGES_REQUESTED** [review]

File: src/cascadia/TerminalApp/TitlebarControl.idl (hunk 1 above)
Comment (AnuthaDev) [reply src/cascadia/TerminalApp/TitlebarControl.idl:11]: Oh, I thought I had fixed that🤔, nvm, doing it asap

**Comment by zadjii-msft** [comment]
> BTW, I would love some guidance on how to compile just the minimum projects necessary for testing UI changes (preferably using VS)

To test these changes, you should be building and running "CascadiaPackage" in VS. That's the project that's the complete Terminal package.

**Pull request closed without merging**